vector-stores/spring-ai-gemfire-store/src/main/java/org/springframework/ai/vectorstore/GemFireVectorStore.java

```diff
@@ -508,4 +508,154 @@ else if (clientException.getStatusCode().equals(BAD_REQUEST)) {
 		}
 	}
 
+	public static class GemFireVectorStoreConfig {
+
+		// Create Index DEFAULT Values
+		public static final String DEFAULT_HOST = "localhost";
+
+		public static final int DEFAULT_PORT = 8080;
+
+		public static final String DEFAULT_INDEX_NAME = "spring-ai-gemfire-index";
+
+		public static final int UPPER_BOUND_BEAM_WIDTH = 3200;
+
+		public static final int DEFAULT_BEAM_WIDTH = 100;
+
+		private static final int UPPER_BOUND_MAX_CONNECTIONS = 512;
+
+		public static final int DEFAULT_MAX_CONNECTIONS = 16;
+
+		public static final String DEFAULT_SIMILARITY_FUNCTION = "COSINE";
+
+		public static final String[] DEFAULT_FIELDS = new String[] {};
+
+		public static final int DEFAULT_BUCKETS = 0;
+
+		public static final boolean DEFAULT_SSL_ENABLED = false;
+
+		String host;
+
+		int port;
+
+		String indexName;
+
+		int beamWidth;
+
+		int maxConnections;
+
+		String vectorSimilarityFunction;
+
+		String[] fields;
+
+		int buckets;
+
+		boolean sslEnabled;
+
+		private GemFireVectorStoreConfig() {
+		}
+
+	}
+
+	/**
+	 * Start building a new configuration.
+	 * @return The entry point for creating a new configuration.
+	 */
+	public static Builder builder() {
+		return new Builder();
+	}
+
+	public static class Builder {
```

Comment:
In many of our other vector store implementations, the builder is provided as an inner class of the configuration. Why do you have it outside of `GemFireVectorStoreConfig`?